libs/pika/async_cuda/include/pika/async_cuda/then_on_host.hpp

```diff
@@ -245,5 +235,16 @@ namespace pika::cuda::experimental {
             return pika::execution::experimental::detail::partial_algorithm<then_on_host_t, F>{
                 PIKA_FORWARD(F, f)};
         }
-    } then_on_host{};
+    };
+
+    /// NOTE: This is not a customization of pika::execution::experimental::then.
+    /// It retains the cuda_scheduler execution context from the predecessor
+    /// sender, but does not run the continuation on a CUDA device. Instead, it
+    /// runs the continuation in the polling thread used by the cuda_scheduler on
+    /// the CPU. The continuation is run only after synchronizing all previous
+    /// events scheduled on the cuda_scheduler. Blocking in the callable given to
+    /// then_on_host blocks other work scheduled on cuda_scheduler from
+    /// completing. Heavier work should be transferred to a host scheduler as
+    /// soon as possible.
+    inline constexpr then_on_host_t then_on_host{};
```

Comment:
I'm considering removing this function as we've so far not had a use for it, and the safer option is to explicitly transfer to a new task if one wants to run something on the host. What do you think about removing it?

Reply:
Remove `then_on_host`.

Reply:
https://github.com/pika-org/pika/pull/1319